apollo-federation/src/query_plan/query_planner.rs

```diff
@@ -103,15 +103,16 @@ pub struct QueryPlannerConfig {
     pub type_conditioned_fetching: bool,
 }
 
+#[allow(clippy::derivable_impls)] // it's derivable right now, but we might change the defaults
 impl Default for QueryPlannerConfig {
     fn default() -> Self {
         Self {
-            reuse_query_fragments: true,
+            reuse_query_fragments: false,
             generate_query_fragments: false,
             subgraph_graphql_validation: false,
             incremental_delivery: Default::default(),
             debug: Default::default(),
-            type_conditioned_fetching: Default::default(),
+            type_conditioned_fetching: false,
```

Comment:
nit: I believe `Default::default()` for boolean is false. Should we update other defaults to explicit false as well (for readability)?

Reply:
Indeed. I wanted to be explicit about this. `incremental_delivery` and `debug` are not booleans but structs, so i kept them as `Default::default`. We could consider manually writing out Default impls for those types.